dask/dataframe/tests/test_dataframe.py

```diff
@@ -2722,3 +2722,61 @@ def test_first_and_last(method):
         for offset in offsets:
             assert_eq(f(ddf, offset), f(df, offset))
             assert_eq(f(ddf.A, offset), f(df.A, offset))
+
+
+@pytest.mark.parametrize('npartitions', [1, 4, 20])
+@pytest.mark.parametrize('split_every', [2, 5])
+@pytest.mark.parametrize('split_out', [None, 1, 5, 20])
+def test_hash_split_unique(npartitions, split_every, split_out):
+    from string import ascii_lowercase
+    s = pd.Series(np.random.choice(list(ascii_lowercase), 1000, replace=True))
+    ds = dd.from_pandas(s, npartitions=npartitions)
+
+    dropped = ds.unique(split_every=split_every, split_out=split_out)
+
+    dsk = dropped._optimize(dropped.dask, dropped._keys())
+    from dask.core import get_deps
+    dependencies, dependents = get_deps(dsk)
+
+    assert len([k for k, v in dependencies.items() if not v]) == npartitions
+    assert dropped.npartitions == (split_out or 1)
+
+
+    assert sorted(dropped.compute()) ==  sorted(s.unique())
+
+
+@pytest.mark.parametrize('npartitions', [1, 4, 20])
+@pytest.mark.parametrize('split_every', [2, 5])
+@pytest.mark.parametrize('split_out', [None, 1, 5, 20])
+def test_hash_groupby_aggregate(npartitions, split_every, split_out):
+    df = pd.DataFrame({'x': np.arange(100) % 10,
+                       'y': np.ones(100)})
+    ddf = dd.from_pandas(df, npartitions)
+
+    result = ddf.groupby('x').y.var(split_every=split_every,
+            split_out=split_out)
+
+    dsk = result._optimize(result.dask, result._keys())
+    from dask.core import get_deps
+    dependencies, dependents = get_deps(dsk)
+
+    assert result.npartitions == (split_out or 1)
+    assert len([k for k, v in dependencies.items() if not v]) == npartitions
+
+    assert_eq(result, df.groupby('x').y.var())
+
+
+def test_hash_split_drop_duplicates():
+    df = pd.DataFrame({'x': [1, 2, 3] * 100})
+    ddf = dd.from_pandas(df, npartitions=5)
+
+    assert ddf.drop_duplicates(split_out=10).npartitions == 10
+    assert_eq(ddf.drop_duplicates(split_out=10), df.drop_duplicates())
+
+
+def test_hash_split_value_counts():
+    df = pd.DataFrame({'x': [1, 2, 3] * 100})
+    ddf = dd.from_pandas(df, npartitions=5)
+
+    assert ddf.x.value_counts(split_out=10).npartitions == 10
+    assert_eq(ddf.x.value_counts(split_out=10), df.x.value_counts())
```

Comment:
General comments for the tests above:
- Slight preference to name with `split_out` (or whatever keyword this ends up being) instead of `hash_split`. It'd be the first thing I'd think of when looking for these tests.
- Slight preference to move the tests for each function next to the existing tests for the functions. The `drop_duplicates` tests are currently all grouped together, as are those for `value_counts` and `unique`.
- Would be nice to have a test that hits both `split_out` and `split_every`.